devices/Ip5306/Ip5306.cs

```diff
@@ -0,0 +1,391 @@
+        /// </summary>
+        public ChargingUnderVoltage ChargingUnderVoltage
+        {
+            get => (ChargingUnderVoltage)(I2cRead(Register.Charger_CTL1) & 0b0001_1100);
+            set
+            {
+                var buf = I2cRead(Register.Charger_CTL1);
+                buf = (byte)(buf & ~0b0001_1100);
+                buf |= (byte)(value);
+                I2cWrite(Register.Charger_CTL1, buf);
+            }
+        }
+
+        /// <summary>
+        /// Charging battery voltage.
+        /// Default to 4.2 Volt.
+        /// </summary>
+        public ChargingBatteryVoltage ChargingBatteryVoltage
+        {
+            get => (ChargingBatteryVoltage)(I2cRead(Register.Charger_CTL2) & 0b0000_1100);
+            set
+            {
+                var buf = I2cRead(Register.Charger_CTL1);
+                buf = (byte)(buf & ~0b0000_1100);
+                buf |= (byte)(value);
+                I2cWrite(Register.Charger_CTL1, buf);
+            }
+        }
+
+        /// <summary>
+        /// Constant charging voltage.
+        /// Default to 14 milli Volt.
+        /// </summary>
+        public ConstantChargingVoltage ConstantChargingVoltage
+        {
+            get => (ConstantChargingVoltage)(I2cRead(Register.Charger_CTL2) & 0b0000_0011);
+            set
+            {
+                var buf = I2cRead(Register.Charger_CTL1);
+                buf = (byte)(buf & ~0b0000_0011);
+                buf |= (byte)(value);
+                I2cWrite(Register.Charger_CTL1, buf);
+            }
+        }
+
+        /// <summary>
+        /// Charging loop selection.
+        /// Defautl to V in.
+        /// </summary>
+        public ChargingLoopSelection ChargingLoopSelection
+        {
+            get => (ChargingLoopSelection)(I2cRead(Register.Charger_CTL3) & 0b0000_1000);
+            set
+            {
+                var buf = I2cRead(Register.Charger_CTL1);
+                buf = (byte)(buf & ~0b0000_1000);
+                buf |= (byte)(value);
+                I2cWrite(Register.Charger_CTL1, buf);
+            }
+        }
+
+        /// <summary>
+        /// Charging current.
+        /// </summary>
+        /// <remarks>Typical valut is between 50 and 3150 milli Ampere. Values are capted to 50 for anything under or 3150 for anything higher.</remarks>
+        public ElectricCurrent ChargingCurrent
+        {
+            get
+            {
+                var buf = I2cRead(Register.CHG_DIG_CTL0) & 0b0001_1111;
+                // I=0.05+b0*0.1+b1*0.2+b2*0.4+b3*0.8+b4*1.6A
+                // I = buff * 100 + 50 mA
+                return ElectricCurrent.FromMilliamperes(buf * 100 + 50);
+            }
+
+            set
+            {
+                // Cap the values to 0 or 3100 mA
+                var currentmA = value.Milliamperes - 50;
+                if (currentmA < 0)
+                {
+                    currentmA = 0;
+                }
+                else if (currentmA > 3100)
```

Comment:
/DRY: declare 3100 once, in a constant.

Reply:
same here, otherwise, you're right, it would be better in a constant if used multiple times